app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -133,6 +138,15 @@ def get_vm_nics(instance)
         end
       end
 
+      def get_nic_security_group(nic)
+        security_group = nic.properties.try(:network_security_group)
+        return unless security_group
+
+        resource_group = security_group.id[%r{resourceGroups/(.*?)/}, 1]
+        security_group_name = File.basename(security_group.id)
+        @nsg.get(security_group_name, resource_group)
+      end
+
```

Comment:
Spoke off line. This PR only gets security_groups associated with NICs. the following changes were discussed:
1) Get all the security_groups in this region
2) Populate the instance :security_groups property by iterating through the instance NICs, grab the security_group ID and use this as a key into the @data_index hash. If it doesn't exist then make an arm-rest API call to get it.
3) Remove the logic that creates then deletes the temporary :security_groups key on the instance NICs